source/extensions/filters/http/transformation/inja_transformer.cc

```diff
@@ -343,45 +374,15 @@ json TransformerInstance::cluster_metadata_callback(const inja::Arguments &args)
     return stringval;
     break;
   }
+  case ProtobufWkt::Value::kStructValue:
+    ABSL_FALLTHROUGH_INTENDED;
   case ProtobufWkt::Value::kListValue: {
-    const auto &listval = value.list_value().values();
-    if (listval.size() == 0) {
-      break;
-    }
-
-    // size is not zero, so this will work
-    auto it = listval.begin();
-    std::stringstream ss;
-
-    auto addValue = [&ss, &it] {
-      const ProtobufWkt::Value &value = *it;
-
-      switch (value.kind_case()) {
-      case ProtobufWkt::Value::kStringValue: {
-        ss << value.string_value();
-        break;
-      }
-      case ProtobufWkt::Value::kNumberValue: {
-        ss << value.number_value();
-        break;
-      }
-      case ProtobufWkt::Value::kBoolValue: {
-        ss << (value.bool_value() ? BoolHeader::get().trueString
-                                  : BoolHeader::get().falseString);
-        break;
-      }
-      default:
-        break;
-      }
-    };
-
-    addValue();
-
-    for (it++; it != listval.end(); it++) {
-      ss << ",";
-      addValue();
-    }
-    return ss.str();
+    std::string output;
+    auto status = value.has_struct_value()
+                      ? ProtobufUtil::MessageToJsonString(value.struct_value(), &output)
```

Comment:
do we have tests to confirm that the final value did not change?

Reply:
What do you mean? For backwards compatibility? I thought you said no one uses this callback

Reply:
hmm i'm not sure what you mean; but yea this might not be needed. let's chat about this